pringles/simulator/simulator.py

```diff
@@ -209,22 +230,41 @@ def run_simulation(self,
                                 output_path=output_path)
 
     @staticmethod
-    def dump_events_in_file(events: List[Event]) -> str:
-        file_descriptor, path = tempfile.mkstemp()
+    def dump_events_in_file(events: List[Event], custom_wd: Optional[str] = None) -> str:
+        path = Simulator._new_temporal_file(custom_wd, "events")
         with open(path, "w") as events_file:
             for event in events:
                 events_file.write(event.serialize() + "\n")
 
-        os.close(file_descriptor)
         return path
 
     @staticmethod
-    def dump_model_in_file(model: Model) -> str:
-        file_descriptor, path = tempfile.mkstemp()
+    def _new_temporal_file(working_dir: Optional[str] = None,
```

Comment:
The name of the method is wrong, it doesn't create temp files if the working dir is set.
Maybe this method should be outside the simulator class.